src/gui/gui.cpp

```diff
@@ -66,22 +70,22 @@ void init()
 				user_campaign_dir, &user_files, &user_dirs, filesystem::name_mode::ENTIRE_FILE_PATH);
 		}
 
-		game_config::config_cache& cache(game_config::config_cache::instance());
 		for(const std::string& umc : user_dirs) {
 			const std::string gui_file = umc + "/gui-theme.cfg";
+			current_file = filesystem::get_short_wml_path(gui_file);
 			if(filesystem::file_exists(gui_file)) {
 				config gui_cfg;
-				cache.get_config(gui_file, gui_cfg);
-				validator.validate(gui_cfg, "gui", 0, gui_file);
+				filesystem::scoped_istream stream = preprocess_file(gui_file, &preproc);
+				read(gui_cfg, *stream, &validator);
 				cfg.append(gui_cfg);
 			}
 		}
 
 	} catch(const config::error& e) {
 		ERR_GUI_P << e.what();
-		ERR_GUI_P << "Setting: could not read file 'data/gui/_main.cfg'.";
+		ERR_GUI_P << "Setting: could not read gui file: " << current_file;
 	} catch(const abstract_validator::error& e) {
-		ERR_GUI_P << "Setting: could not read file 'data/schema/gui.cfg'.";
+		ERR_GUI_P << "Setting: could not read schema file: " << schema_file;
```

Comment:
Though I don't have a preference about this change, it's technically not necessary, is it? Since all these files do use the same schema. It's not like it can hurt though…